runtime/src/main/java/org/corfudb/runtime/collections/StreamPollingTask.java

```diff
@@ -59,7 +60,7 @@ public void run() {
             log.error("Encountered exception {} during txn stream polling, listener: {}, " +
                     "namespace: {}", throwable, listener, subscription.getNamespace());
             streamingManager.unsubscribe(listener, false);
-            listener.onError(throwable);
+            listener.onError(new StreamingException(throwable));
```

Comment:
This will also propagate `RejectedExecutionException` and `InterruptedException` to the listener, but it shouldn't right? If those happen the client cannot react to those exceptions and we manage those threads. What is the benefit of signaling an exception on our threads to the consumer listeners?

Reply:
But then how do they know they're no longer subscribed, and deltas won't be coming in any more? the case of the RejectedExecutionException, deltas won't be coming in...